contracts/core/EntryPoint.sol

```diff
@@ -240,8 +239,34 @@ contract EntryPoint is IEntryPoint, StakeManager {
      * @return sigForAggregation  - only if has actualAggregator:  this value is returned from IAggregator.validateUserOpSignature, and should be passed to aggregator.aggregateSignatures
      * @return offChainSigInfo - if has actualAggregator, and offChainSigCheck is true, this value should be used by the off-chain signature code (e.g. it contains the sender's publickey)
      */
-    function simulateValidation(UserOperation calldata userOp, bool offChainSigCheck)
+    function simulateValidation(UserOperation calldata userOp)
     external returns (uint256 preOpGas, uint256 prefund, address actualAggregator, bytes memory sigForUserOp, bytes memory sigForAggregation, bytes memory offChainSigInfo) {
+        address[] memory allowedAggregators = new address[](1);
+        allowedAggregators[0] = address (0);
+        return simulateValidationWithAggregators(userOp, allowedAggregators, false);
+    }
+
+    /**
+    * Simulate a call to wallet.validateUserOp and paymaster.validatePaymasterUserOp.
+    * allow using a sighnature aggregator.
+    * Validation succeeds if the call doesn't revert.
+    * @dev The node must also verify it doesn't use banned opcodes, and that it doesn't reference storage outside the wallet's data.
+     *      In order to split the running opcodes of the wallet (validateUserOp) from the paymaster's validatePaymasterUserOp,
+     *      it should look for the NUMBER opcode at depth=1 (which itself is a banned opcode)
+     * @param userOp the user operation to validate.
+     * @param allowedAggregators list of aggregators we specifically allow for this userOp. An empty list means "allow any".
+     *          a list of a single "address(0)" entry means "do not allow any other aggregator", effectively means blocking aggregated signatures.
+     * @param offChainSigCheck if the wallet has an aggregator, skip on-chain aggregation check. In thus case, the bundler must
+     *          perform the equivalent check using an off-chain library code
+     * @return preOpGas total gas used by validation (including contract creation)
+     * @return prefund the amount the wallet had to prefund (zero in case a paymaster pays)
+     * @return actualAggregator the aggregator used by this userOp. if a non-zero aggregator is returned, the bundler must get its params using
+     *      aggregator.
+     * @return sigForUserOp - only if has actualAggregator: this value is returned from IAggregator.validateUserOpSignature, and should be placed in the userOp.signature when creating a bundle.
+     * @return sigForAggregation  - only if has actualAggregator:  this value is returned from IAggregator.validateUserOpSignature, and should be passed to aggregator.aggregateSignatures
+     * @return offChainSigInfo - if has actualAggregator, and offChainSigCheck is true, this value should be used by the off-chain signature code (e.g. it contains the sender's publickey)
```

Comment:
Why do we need `offChainSigInfo` rather than just reuse `sigForAggregation`?  Does the off-chain aggregation implementation need different information than the on-chain aggregation code?

Reply:
Never mind, I just saw you replied this question on the other PR.